base/env.jl

```diff
@@ -98,6 +98,38 @@ Base.EnvDict with "N" entries:
 """
 const ENV = EnvDict()
 
+"""
+    Base.get_bool_env(name::String, default::Bool = false)::Bool
+
+Evaluate whether the value of environnment variable `name` is truthy or falsy,
+and error if none are recognized. If the variable is not set, or is set to "",
+return `default`.
+
+Recognized values are:
+    truthy: "t", "true", "y", "yes", "1"
+    falsy:  "f", "false", "n", "no", "0"
+"""
+function get_bool_env(name::String, default::Bool = false)
+    haskey(ENV, name) || return default
+    val = lowercase(get(ENV, name, ""))
```

Comment:
probably this doesn't matter, as this won't be called in a performance critical place, but... maybe we should avoid this allocating when possible, by checking if the string is already lowercase (which i think should be okay overhead in practice since we're not expecting really long strings here)
```suggestion
    val = get(ENV, name, "")
    val = all(islowercase, va) ? val : lowercase(val)
```
idk, just an thought

Reply:
oh on the other hand this is a whole can of worms so maybe ignore this
https://github.com/JuliaLang/julia/issues/46444

Reply:
If calling `lowercase` is worrying, adding the explicit list in `truthy` is also an option, at least if restricted to all lower or upper cases (i.e. not including `"TrUe"`...).